test/common/export_results.c

```diff
@@ -98,6 +131,13 @@ int test_common_options(test_common_options_t *options)
 		return -1;
 	}
 
+	if (gbl_data.args) {
+		fprintf(gbl_data.file, "#%.*s;%s\n", (int)filename_len, gbl_data.filename,
```

Comment:
Could add check for `fprintf()` failures.  Also, application could at least in theory call `test_common_options()` multiple times, which would currently cause problems. An internal variable e.g. `init_done` could be used to track this, so file open and header write would be done only once. Otherwise the PR look OK to me.